crates/node/p2p/src/peers/utils.rs

```diff
@@ -17,3 +19,76 @@ pub fn enr_to_multiaddr(enr: &Enr) -> Option<Multiaddr> {
     }
     None
 }
+
+/// This tag should be set to indicate to libsecp256k1 that the following bytes denote an
+/// uncompressed pubkey.
+///
+/// `SECP256K1_TAG_PUBKEY_UNCOMPRESSED` = `0x04`
+///
+/// See: <https://github.com/bitcoin-core/secp256k1/blob/master/include/secp256k1.h#L211>
+const SECP256K1_TAG_PUBKEY_UNCOMPRESSED: u8 = 4;
+
+/// Converts an uncompressed [`PeerId`] to a [`secp256k1::PublicKey`] by prepending the [`PeerId`]
+/// bytes with the `SECP256K1_TAG_PUBKEY_UNCOMPRESSED` tag.
+#[inline]
+pub fn id2pk(id: PeerId) -> Result<secp256k1::PublicKey, secp256k1::Error> {
+    // NOTE: B512 is used as a PeerId because 512 bits is enough to represent an uncompressed
+    // public key.
+    let mut s = [0u8; secp256k1::constants::UNCOMPRESSED_PUBLIC_KEY_SIZE];
+    s[0] = SECP256K1_TAG_PUBKEY_UNCOMPRESSED;
+    s[1..].copy_from_slice(id.as_slice());
+    secp256k1::PublicKey::from_slice(&s)
+}
+
+/// Converts an uncoded [`PeerId`] to a [`libp2p::PeerId`].
+pub fn local_id_to_p2p_id(peer_id: PeerId) -> Result<libp2p::PeerId, secp256k1::Error> {
+    // The libp2p library works with compressed public keys.
+    let encoded_pk_bytes = id2pk(peer_id)?.serialize();
+    let pk: discv5::libp2p_identity::PublicKey =
+        discv5::libp2p_identity::secp256k1::PublicKey::try_from_bytes(&encoded_pk_bytes)
+            .expect("secp256k1 serialization should be valid")
+            .into();
+
+    Ok(pk.to_peer_id())
+}
```

Comment:
make sure these functions are full test covered, i.e. all the error branches, seems like code cov is happy with them already :)